src/main/java/org/openjdk/jextract/impl/TreeMaker.java

```diff
@@ -289,10 +289,10 @@ final Type.Declared recordDeclaration(Cursor parent, Cursor recordCursor) {
                     if (fc.isAnonymousStruct()) {
                         // process struct recursively
                         pendingFields.add(recordDeclaration(parent, fc).tree());
-                    } else {
+                    } else if (fc.kind() == CursorKind.FieldDecl) {
```

Comment:
This extra check seems not very useful. Note that this section is guarded by "isFlattenable" which will exclude attributes, and stuff that we do not support.